src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -76,103 +107,126 @@
     public abstract List<String> getPackageNames();
 
     /**
-     * Give this descriptor a chance to filter out any classes it doesn't want to be
-     * dynamically discoverable.
-     * @return false if the class shouldn't be used; otherwise true
+     * Base class for all command line plugin classes managed by this descriptor. Subclasses of
+     * this class in any of the packages returned by {@link #getPackageNames} will be command line
+     * accessible.
+     */
+    public abstract Class<?> getPluginBaseClass();
+
+    /**
+     * Determine if a plugin class should be retained for this descriptor. Return true if the
+     * descriptor wants this class to be included in from the list of plugins discovered dynamically.
+     *
+     * @return false if the plugin class shouldn't be used; otherwise true
      */
-    public Predicate<Class<?>> getClassFilter() { return c -> true;}
+    public boolean includePluginClass(Class<?> c) { return true;}
```

Comment:
I'm reluctant to make that change. Unless there is a bug somewhere, isAssignableFrom should never return false. Changing the implementation to call it would silently mask such an issue.